### PR TITLE
Add a set of file-related utilities that will be used in corpus traversal during replay.

### PR DESCRIPTION
**Pull request by stefanbucur**

### CODE REVIEW

File: fuzzing/replay/file_util.cc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    const std::string& path,
+    absl::FunctionRef<void(absl::string_view, const struct stat&)> callback) {
+  struct stat path_stat;
+  if (stat(path.c_str(), &path_stat) < 0) {
+    return ErrnoStatus(absl::StrCat("could not stat ", path), errno);
+  }
+  if (S_ISDIR(path_stat.st_mode)) {
+    return TraverseDirectory(path, callback);
+  } else {
+    callback(path, path_stat);
```
Comment (oliverchang) [inline fuzzing/replay/file_util.cc:78]: nit: unnecessary else after return.

File: fuzzing/replay/file_util.cc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  FILE* f = fopen(path.c_str(), "w");
+  if (!f) {
+    return ErrnoStatus("could not open file", errno);
+  }
+  const size_t result = fwrite(contents.data(), 1, contents.size(), f);
+  fclose(f);
+  if (result < contents.size()) {
+    return absl::UnknownError("could not write file contents");
+  } else {
+    return absl::OkStatus();
```
Comment (oliverchang) [inline fuzzing/replay/file_util.cc:94]: nit: unnecessary else after return.

**Review by oliverchang — APPROVED** [review]

File: fuzzing/replay/file_util.cc (hunk 1 above)
Comment (stefanbucur) [reply fuzzing/replay/file_util.cc]: Fixed.

File: fuzzing/replay/file_util.cc (hunk 2 above)
Comment (stefanbucur) [reply fuzzing/replay/file_util.cc]: Fixed.

**Pull request merged**